conans/test/functional/toolchains/cmake/test_cmaketoolchain_paths.py

```diff
@@ -0,0 +1,57 @@
+import textwrap
+
+import pytest
+
+from conans.test.utils.tools import TestClient
+
+
+@pytest.mark.tool_cmake
+@pytest.mark.parametrize("package", ["hello", "ZLIB"])
+@pytest.mark.parametrize("find_package", ["module", "config"])
+def test_cmaketoolchain_path_find(package, find_package):
+    """Test with user "Hello" and also ZLIB one, to check that package ZLIB
+    has priority over the CMake system one
+
+    Also, that user cmake files in the root are accessible via CMake include()
+    """
+    client = TestClient()
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        class TestConan(ConanFile):
+            exports = "*"
+            def layout(self):
+                pass
+            def package(self):
+                self.copy(pattern="*", keep_path=False)
+        """)
```

Comment:
Missing `cpp_info.buildirs = "."` ? I would say it shouldn't work without it.

Reply:
Test passes green, I thought that too initially... 🤷 😄

Reply:
We have to check why

Reply:
I'll check, yes

Reply:
It seems it makes sense:

- ``CppInfo`` declares a ``builddirs = ""``
- ``NewCppInfo`` inherits that value
- ``CMakeToolchain`` generates ``CMAKE_MODULE_PATH``, ``CMAKE_PREFIX_PATH`` from ``for p in cppinfo.builddirs``, doing a os.path.join(), pointing then to the root of the package
- The test is putting the find and config files in the root of the package (many packages will do that)

Reply:
that makes sense. I thought the default was `[]`. Actually, I don't think the default should be the root... why the root? why by default? Do we want that for Conan 2.0?

Reply:
Because it is common to create xxx-config.cmake files or find modules in the root of the package, I guess.

What I am concerned here is the transition, does it mean that recipes should add in Conan 1.X all the ``includedirs``, ``libdirs`` directories to their ``package_info()`` now, because there are no defaults for any of them? Lets discuss it.

Reply:
Lets check what is the default of cmake.install() generating a config.cmake file.

Reply:
I have added a real helloConfig.cmake. There is no real default in cmake, and it will look in many different locations: https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure. Libraries CMakeLists.txt will define different locations, but none of them seems to make sense as a default for Conan. In the test I have used ``hello/cmake`` as copied from the internet. The only possibility I see is the root of the package (``CMAKE_INSTALL_PREFIX``)